<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MiningFrame</class>
 <widget class="QFrame" name="MiningFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1195</width>
    <height>547</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::NoFrame</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Raised</enum>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,1,0">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="m_miningManageFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>100</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>100</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="leftMargin">
       <number>25</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>25</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <widget class="WalletGui::WalletNormalGrayTextLabel" name="label">
        <property name="text">
         <string>CPU CORES:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="m_cpuCoresCombo"/>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>13</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="WalletGui::WalletNormalGrayTextLabel" name="label_2">
        <property name="text">
         <string>POOL SELECTION STRATEGY:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_failoverStrategyRadio">
        <property name="toolTip">
         <string>Starts mining on the first pool. If it fails chooses the next one.</string>
        </property>
        <property name="text">
         <string>Failover</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">m_poolSelectionStrategyButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_randomStrategyRadio">
        <property name="toolTip">
         <string>Chooses random pool from the pool list.</string>
        </property>
        <property name="text">
         <string>Random</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">m_poolSelectionStrategyButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>13</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QCheckBox" name="m_lockedScreenMiningCheck">
        <property name="text">
         <string>Start mining when locked</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>31</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QFrame" name="m_hashRateFrame">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>5</number>
         </property>
         <item row="2" column="0">
          <widget class="WalletGui::WalletNormalBlackTextLabel" name="m_donationHashRateTextLabel">
           <property name="text">
            <string>Donation:</string>
           </property>
           <property name="indent">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="1" alignment="Qt::AlignLeft">
          <widget class="WalletGui::WalletNormalBlackTextLabel" name="m_donationHashRateLabel">
           <property name="text">
            <string/>
           </property>
           <property name="indent">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1" alignment="Qt::AlignLeft">
          <widget class="WalletGui::WalletHeaderLabel" name="m_hashrateLabel">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="WalletGui::WalletHeaderLabel" name="label_7">
           <property name="text">
            <string>Hashrate:</string>
           </property>
           <property name="indent">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>13</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="WalletGui::WalletNormalBlueButton" name="m_startMiningButton">
        <property name="minimumSize">
         <size>
          <width>190</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>190</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Start mining</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="leftMargin">
      <number>25</number>
     </property>
     <property name="topMargin">
      <number>20</number>
     </property>
     <property name="rightMargin">
      <number>25</number>
     </property>
     <property name="bottomMargin">
      <number>20</number>
     </property>
     <item>
      <widget class="WalletGui::WalletHeaderLabel" name="label_6">
       <property name="text">
        <string>Mining pool list</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="leftMargin">
      <number>25</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>25</number>
     </property>
     <property name="bottomMargin">
      <number>20</number>
     </property>
     <item>
      <widget class="WalletGui::PoolTreeView" name="m_poolView">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContents</enum>
       </property>
       <property name="dragEnabled">
        <bool>true</bool>
       </property>
       <property name="dragDropOverwriteMode">
        <bool>false</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::InternalMove</enum>
       </property>
       <property name="defaultDropAction">
        <enum>Qt::MoveAction</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="rootIsDecorated">
        <bool>false</bool>
       </property>
       <property name="uniformRowHeights">
        <bool>true</bool>
       </property>
       <property name="itemsExpandable">
        <bool>false</bool>
       </property>
       <property name="allColumnsShowFocus">
        <bool>true</bool>
       </property>
       <attribute name="headerStretchLastSection">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QFrame" name="m_addPoolFrame">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="leftMargin">
       <number>25</number>
      </property>
      <property name="topMargin">
       <number>23</number>
      </property>
      <property name="rightMargin">
       <number>25</number>
      </property>
      <property name="bottomMargin">
       <number>23</number>
      </property>
      <item row="0" column="0">
       <widget class="WalletGui::WalletNormalGrayTextLabel" name="label_3">
        <property name="text">
         <string>HOST</string>
        </property>
        <property name="indent">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="WalletGui::WalletNormalGrayTextLabel" name="label_4">
        <property name="text">
         <string>PORT</string>
        </property>
        <property name="indent">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="WalletGui::WalletNormalGrayTextLabel" name="label_5">
        <property name="text">
         <string>DIFFICULTY</string>
        </property>
        <property name="indent">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="m_newPoolHostEdit">
        <property name="placeholderText">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="m_newPoolPortSpin">
        <property name="specialValueText">
         <string/>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSpinBox" name="m_difficultySpin">
        <property name="specialValueText">
         <string/>
        </property>
        <property name="maximum">
         <number>999999999</number>
        </property>
        <property name="singleStep">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="WalletGui::WalletNormalBlueButton" name="m_addPoolButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>80</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Add pool</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WalletGui::WalletNormalBlueButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletBlueButton.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletHeaderLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletNormalGrayTextLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletNormalBlackTextLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::PoolTreeView</class>
   <extends>QTreeView</extends>
   <header>Gui/Mining/PoolTreeView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_poolSelectionStrategyButtonGroup</sender>
   <signal>buttonToggled(QAbstractButton*,bool)</signal>
   <receiver>MiningFrame</receiver>
   <slot>schedulePolicyChanged(QAbstractButton*,bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_startMiningButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MiningFrame</receiver>
   <slot>startMiningClicked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1012</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_newPoolHostEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MiningFrame</receiver>
   <slot>newPoolUrlChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>409</x>
     <y>348</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_newPoolPortSpin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MiningFrame</receiver>
   <slot>newPoolUrlChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>844</x>
     <y>348</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_addPoolButton</sender>
   <signal>clicked()</signal>
   <receiver>MiningFrame</receiver>
   <slot>addPoolClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1067</x>
     <y>348</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_poolView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>MiningFrame</receiver>
   <slot>deleteClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_cpuCoresCombo</sender>
   <signal>activated(int)</signal>
   <receiver>MiningFrame</receiver>
   <slot>cpuCoreCountChangedByUser(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_lockedScreenMiningCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>MiningFrame</receiver>
   <slot>miningOnLockedScreenChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>728</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startMiningClicked(bool)</slot>
  <slot>addPoolClicked()</slot>
  <slot>newPoolUrlChanged()</slot>
  <slot>schedulePolicyChanged(QAbstractButton*,bool)</slot>
  <slot>deleteClicked(QModelIndex)</slot>
  <slot>cpuCoreCountChangedByUser(int)</slot>
  <slot>miningOnLockedScreenChecked(bool)</slot>
 </slots>
 <buttongroups>
  <buttongroup name="m_poolSelectionStrategyButtonGroup"/>
 </buttongroups>
</ui>
